storehaus-postgresql/src/main/scala/com/twitter/storehaus/postgres/package.scala

```diff
@@ -1,60 +1,69 @@
 package com.twitter.storehaus
 
+import com.twitter.bijection.Injection
 import roc.postgresql.ElementDecoder
 import roc.types.failures.NullDecodedFailure
 
+
+import scala.util.Try
+
 /**
   * Created by ponkin on 10/24/16.
   */
 package object postgres {
 
+  type PostgresValueConverter[A] = Injection[A, PostgresValue]
+
   import roc.types.decoders._
 
-  // Read instances
+  trait OneWayInjection[T] extends Injection[T, PostgresValue] {
+    override def apply(t: T): PostgresValue =
+      throw IllegalConversionException(s"Can`t convert ${t.getClass.getCanonicalName} to PostgresValue")
+  }
 
-  implicit val boolRead = new Read[Boolean] {
-    override def read(v: PostgresValue): Boolean = v match {
-      case RocPostgresValue(e) => e.as[Boolean]
+  implicit val boolInjection= new OneWayInjection[Boolean] {
```

Comment:
okay, I'm really sorry. I overlooked that your `Read` and `Show` were not inverses of each other. One is going to `String` the other is coming from a PostgresValue. I wish I had noticed that.

I'd actually prefer the way you had it to throwing exceptions. I mistakenly thought we could combine `Read` and `Show` into `Injection` but not the way you have the code.

What about a direct trait like:
```scala
trait PostgresCodec[A] {
  def encode(a: A): String
  def decode(p: PostgresValue): Try[A]
}
```
I don't see a need to decouple these since they are so special purpose. I thought Injection would do, but the types are not the same on encode and decode.